### ISSUE TITLE
Insert Stages at Any Point in Pipelines

### ISSUE THREAD

**Issue by MarianRaphael** [flowforge/flowforge#2299]
### Description

Currently, our pipeline only supports adding stages to the end. While this might cover many use cases, there are scenarios in which users might need to insert a stage before the current ones (for example, adding a 'staging' stage before 'prod' in an existing application). Furthermore, there may be cases where we might need to add a new stage in the middle of two existing stages.

Having the flexibility to add a stage anywhere in the pipeline not only mirrors real-world software development workflows but also provides users with greater control over their development, testing, and deployment processes.

### Have you provided an initial effort estimate for this issue?

I have provided an initial effort estimate

**Comment by Steve-Mcl** [comment]
If the effort to achieve this FR is the same (or less than), I would like to propose this is achieved by drag + drop. 

e.g. Add a stage, drag it into position.

This means if you need to re-order stages, there is no need to delete and re-create the "out-of-place" item.

PS: Drag+Drop can be substituted with "promote/demote" buttons if implementation of drag+drop is deemed not UX friendly or too much effort.

**Comment by joepavitt** [comment]
Curious @MarianRaphael has this been customer feedback/request at all?

**Comment by joepavitt** [comment]
A lot of complexities here to consider (particular for something as free as drag & drop) is that we have different stages that have different constraints depending on where they are in the pipeline, and what the succeed and precede.

**Comment by MarianRaphael** [comment]
> Curious @MarianRaphael has this been customer feedback/request at all?

No, this was raised as an outcome of an internal discussion. No customer requested this yet.
